include/response_reader.rb

```diff
@@ -105,10 +105,8 @@ def url_and_args_from_location
     url << "?#{uri.query}" if uri.query.to_s.length > 0
     url = url.gsub(/\A\//, "")
 
-    args = {host: uri.host}
+    args = @http2.args.merge(host: uri.host)
```

Comment:
This will also clone the port, which I guess it shouldn't since it should use the port from the given location?